WordPressKitTests/WordPressComRestApiTests.swift

```diff
@@ -60,6 +60,66 @@ class WordPressComRestApiTests: XCTestCase {
         }
     }
 
+    @available(iOS 16.0, *)
+    func testQuery() {
+        var requestURL: URL?
+        stub(condition: isRestAPIRequest()) {
+            requestURL = $0.url
+            return HTTPStubsResponse(error: URLError(URLError.Code.networkConnectionLost))
+        }
+
+        let expect = self.expectation(description: "One callback should be invoked")
+        let api = WordPressComRestApi(oAuthToken: "fakeToken")
+        api.GET(
+            wordPressMediaRoutePath,
+            parameters: [
+                "number": 1,
+                "true": true,
+                "false": false,
+                "string": "true",
+                "dict": ["foo": true, "bar": "string"],
+                "nested-dict": [
+                    "outter1": [
+                        "inner1": "value1",
+                        "inner2": "value2"
+                    ],
+                    "outter2": [
+                        "inner1": "value1",
+                        "inner2": "value2"
+                    ]
+                ],
+                "array": ["true", 1, false]
+            ] as [String: AnyObject],
+            success: { _, _ in expect.fulfill() },
+            failure: { (_, _) in expect.fulfill() }
+        )
+        wait(for: [expect], timeout: 0.1)
+
+        let query = requestURL?
+            .query(percentEncoded: false)?
+            .split(separator: "&")
+            .reduce(into: Set()) { $0.insert(String($1)) }
+        let expected = [
+            "number=1",
+            "true=1",
+            "false=0",
+            "string=true",
+            "dict[foo]=1",
+            "dict[bar]=string",
+            "nested-dict[outter1][inner1]=value1",
+            "nested-dict[outter1][inner2]=value2",
+            "nested-dict[outter2][inner1]=value1",
+            "nested-dict[outter2][inner2]=value2",
+            "array[]=true",
+            "array[]=1",
+            "array[]=0",
+        ]
+
+        for item in expected {
+            XCTAssertTrue(query?.contains(item) == true, "Missing query item: \(item)")
+        }
```

Comment:
After checking that all the expected items are in the query, how do you feel about checking _only_ the expected items are in the query?

```suggestion
        }
        
        XCTAssertEqual(expected.count, query.count)
```